tools/dm-task-config-argument-description.md

```diff
@@ -0,0 +1,203 @@
+---
+title: Task Configuration Argument Description of Data Migration
+summary: This document introduces some arguments in task configuration of Data Migration.
+category: tools
+---
+
+# Task Configuration Argument Description of Data Migration
+
+This document introduces some arguments in task configuration of Data Migration. 
+
+## `task-mode`
+
+- String
+- The task mode of data migration to be executed 
+- Value: `full`/`incremental`/`all`
+
+    - `full`: Only makes a full backup of the upstream database and then makes a full import to the downstream database.
+    - `incremental`: Only synchronizes the incremental data of the upstream database to the downstream database using the binlog.
+    - `all`: `full` + `incremental`. Makes a full backup of the upstream database, imports the full data to the downstream database, and then uses the binlog to make an incremental synchronization to the downstream database starting from the exported position during the full backup process (binlog position/GTID).
+
+## Routing rule
+
+```
+# `schema-pattern`/`table-pattern` uses the wildcard matching rule
+schema level:
+​    schema-pattern: "test_*"
+​    target-schema: "test"
+
+table level:
+​    schema-pattern: "test_*"
+​    table-pattern: "t_*"
+​    target-schema: "test"
+​    target-table: "t"
+```
+
+Description: Synchronizes the upstream table data that matches `schema-pattern`/`table-pattern` to the downstream `target-schema`/`target-table`. You can set the routing rule at the schema/table level. 
+
+Taking the above code block as an example:
+
+- Schema level: Synchronizes all the upstream tables that match the `test_*` schema to the downstream `test` schema.  
+
+    For example, `schema: test_1 - tables [a, b, c]`  =>  `schema:test -  tables [a, b, c]`
+
+- Table level: Synchronizes the `t_*` matched upstream tables with `test_*` matched schema to the downstream `schema:test table:t` table.
+
+> **Notes:** 
+>
+> - The `table level` rule has a higher priority than the `schema level` rule.
+> - You can set one configuration rule at most at one level.
+
+
+## Black white list rule
+
+```
+instance:						
+​    do-dbs: ["~^test.*", "do"]         # Starts with "~", indicating it is a regular expression
+​    ignore-dbs: ["mysql", "ignored"]
+​    do-tables:
+​    - db-name: "~^test.*"
+​      tbl-name: "~^t.*"
+​    - db-name: "do"
+​      tbl-name: "do"
+​    ignore-tables:
+​    - db-name: "do"
+​      tbl-name: "do"
+```
+
+Description: The black white list filter rule for the upstream database instance table. This rule is similar to `replication-rules-db`/`replication-rules-table` of MySQL.
```

Comment:
MySQL replication rules for replicate-do-db are very complicated :(  I would be careful just saying it is the same: https://dev.mysql.com/doc/refman/8.0/en/replication-options-slave.html#option_mysqld_replicate-do-db

It is likely the same when in RBR, but old DBAs will be familiar with SBR which has strange semantics (different DB selected, transaction spanning dbs).

Reply:
@morgo I will remove "This rule is similar to `replication-rules-db`/`replication-rules-table` of MySQL."